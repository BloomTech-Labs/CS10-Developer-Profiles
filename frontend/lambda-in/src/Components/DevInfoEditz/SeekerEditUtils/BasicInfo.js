import TextField from '@material-ui/core/TextField';
import React from 'react';
import SEEKER_PROP_TYPES from '../../seekerPropTypes';
import StateCapsule from '../../utilityComponents/StateCapsule/StateCapsule';

const BasicInfo = ({ userInfo }) => {
  const schema = {
    firstName: userInfo.firstName,
    lastName: userInfo.lastName,
    desiredTitle: userInfo.desiredTitle,
    currentLocation: userInfo.currentLocation,
  };

  return (
    <StateCapsule schema={schema} object={schema}>
      {({ stateCapsule }) => (
        <div className="smallInputContainer">
          <div className="expansionPanel">
            <div className="inputField">
              <TextField
                id="edit-firstName"
                label="First Name"
                fullWidth
                value={stateCapsule.firstName}
                margin="normal"
                variant="outlined"
              />
            </div>

            <div className="inputField">
              <TextField
                id="edit-lastName"
                label="Last Name"
                fullWidth
                value={stateCapsule.lastName}
                margin="normal"
                variant="outlined"
              />
            </div>

            <div className="inputField">
              <TextField
                id="edit-desiredTitle"
                label="Desired Title"
                fullWidth
                value={stateCapsule.desiredTitle}
                margin="normal"
                variant="outlined"
              />
            </div>

            {/* TODO : WIRE InputGeolocation component here */}
            {/* <div className="inputField">
              <TextField
                id="edit-currentLocation"
                label="Current Location"
                fullWidth
                value={stateCapsule.currentLocation}
                margin="normal"
                variant="outlined"
              />
            </div> */}
          </div>
          <div className="imageContainer">
            <img
              className="displayPic"
              src={`https://robohash.org/1${stateCapsule.firstName}`}
              alt="Italian "
            />
          </div>
        </div>
      )}
    </StateCapsule>
  );
};

BasicInfo.propTypes = {
  userInfo: SEEKER_PROP_TYPES.isRequired,
};

export default BasicInfo;
