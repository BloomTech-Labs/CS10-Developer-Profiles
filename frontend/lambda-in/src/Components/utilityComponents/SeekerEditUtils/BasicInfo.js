import React, { Fragment } from 'react';
import TextField from '@material-ui/core/TextField';
import StateCapsule from '../StateCapsule/StateCapsule';

// eslint-disable-next-line arrow-parens
const BasicInfo = props => {
  const { userInfo } = props;

  const schema = {
    firstName: userInfo.firstName,
    lastName: userInfo.lastName,
    desiredTitle: userInfo.desiredTitle,
    currentLocation: userInfo.currentLocation,
  };

  return (
    <Fragment>
      <StateCapsule schema={schema} object={schema}>
        {({ stateCapsule }) => (
          <div className="smallInputContainer">
            <div className="expansionPanel">
              <div className="inputField">
                <TextField
                  id="new-firstName"
                  label="First Name"
                  fullWidth
                  value={stateCapsule.firstName}
                  margin="normal"
                  variant="outlined"
                />
              </div>

              <div className="inputField">
                <TextField
                  id="new-lastName"
                  label="Last Name"
                  fullWidth
                  value={stateCapsule.lastName}
                  margin="normal"
                  variant="outlined"
                />
              </div>

              <div className="inputField">
                <TextField
                  id="new-desiredTitle"
                  label="Desired Title"
                  fullWidth
                  value={stateCapsule.desiredTitle}
                  margin="normal"
                  variant="outlined"
                />
              </div>

              {/* TODO : WIRE InputGeolocation component here */}
              <div className="inputField">
                <TextField
                  id="new-currentLocation"
                  label="Current Location"
                  fullWidth
                  value={stateCapsule.currentLocation}
                  margin="normal"
                  variant="outlined"
                />
              </div>
            </div>
            <div className="imageContainer">
              <img
                className="displayPic"
                src={`https://robohash.org/1${stateCapsule.firstName}`}
                alt="Italian "
              />
            </div>
          </div>
        )}
      </StateCapsule>
    </Fragment>
  );
};

export default BasicInfo;
