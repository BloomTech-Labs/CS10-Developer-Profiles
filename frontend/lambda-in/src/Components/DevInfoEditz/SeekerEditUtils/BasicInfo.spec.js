import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BasicInfo from './BasicInfo';

configure({ adapter: new Adapter() });

/**
 * @author jesuarva
 * We are only testing proper rendering of inner components,
 * this is accomplish with a snapshot.
 *
 * @see StateCapsule.spec.js for onChange functionality tests.
 * @todo Wire InputGeolocation.js component and test is rendering.
 */
describe('BasicInfo component', () => {
  const userInfo = {
    firstName: 'Happy',
    lastName: 'User',
    desiredTitle: 'Software developer',
    currentLocation: 'Madrid',
  };
  const basicInfo = <BasicInfo userInfo={userInfo} />;

  it('should renders correctly', () => {
    const tree = renderer.create(basicInfo);
    expect(tree).toMatchSnapshot();
  });
});
