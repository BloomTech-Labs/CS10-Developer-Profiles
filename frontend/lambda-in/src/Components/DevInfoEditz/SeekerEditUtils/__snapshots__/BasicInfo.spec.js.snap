// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo component should renders correctly 1`] = `
<div
  className="state-capsule"
  onChange={[Function]}
  onKeyPress={[Function]}
>
  <div
    className="smallInputContainer"
  >
    <div
      className="expansionPanel"
    >
      <div
        className="inputField"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <label
            className="MuiFormLabel-root-12 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8 MuiInputLabel-outlined-11"
            data-shrink={true}
            htmlFor="edit-firstName"
          >
            First Name
          </label>
          <div
            className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-fullWidth-41 MuiInputBase-formControl-33"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiNotchedOutline-legend-50"
                style={
                  Object {
                    "width": NaN,
                  }
                }
              />
            </fieldset>
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input-42 MuiOutlinedInput-input-27"
              defaultValue={undefined}
              disabled={false}
              id="edit-firstName"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value="Happy"
            />
          </div>
        </div>
      </div>
      <div
        className="inputField"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <label
            className="MuiFormLabel-root-12 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8 MuiInputLabel-outlined-11"
            data-shrink={true}
            htmlFor="edit-lastName"
          >
            Last Name
          </label>
          <div
            className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-fullWidth-41 MuiInputBase-formControl-33"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiNotchedOutline-legend-50"
                style={
                  Object {
                    "width": NaN,
                  }
                }
              />
            </fieldset>
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input-42 MuiOutlinedInput-input-27"
              defaultValue={undefined}
              disabled={false}
              id="edit-lastName"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value="User"
            />
          </div>
        </div>
      </div>
      <div
        className="inputField"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <label
            className="MuiFormLabel-root-12 MuiFormLabel-filled-16 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8 MuiInputLabel-outlined-11"
            data-shrink={true}
            htmlFor="edit-desiredTitle"
          >
            Desired Title
          </label>
          <div
            className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-fullWidth-41 MuiInputBase-formControl-33"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiNotchedOutline-legend-50"
                style={
                  Object {
                    "width": NaN,
                  }
                }
              />
            </fieldset>
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input-42 MuiOutlinedInput-input-27"
              defaultValue={undefined}
              disabled={false}
              id="edit-desiredTitle"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value="Software developer"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="imageContainer"
    >
      <img
        alt="Italian "
        className="displayPic"
        src="https://robohash.org/1Happy"
      />
    </div>
  </div>
</div>
`;
